tests/functional/olp-cpp-sdk-dataservice-read/DataserviceReadVolatileLayerClientTest.cpp

```diff
@@ -224,4 +341,244 @@ TEST_F(VolatileLayerClientTest, GetPartitionsDifferentFetchOptions) {
+    DataResponse response;
+    auto token = client.GetData(
+        olp::dataservice::read::DataRequest()
+            .WithPartitionId(sub_partition2)
+            .WithFetchOption(CacheOnly),
+        [&response](DataResponse resp) { response = std::move(resp); });
+    EXPECT_TRUE(response.IsSuccessful());
+    ASSERT_TRUE(response.GetResult() != nullptr);
+    ASSERT_NE(response.GetResult()->size(), 0u);
+  }
+}
+
+TEST_F(VolatileLayerClientTest, PrefetchWrongLevels) {
+  const auto kTileId = GetTestTile();
+  auto layer = CustomParameters::getArgument(kPrefetchLayer);
+  auto catalog = CustomParameters::getArgument(kPrefetchCatalog);
+
+  PrepareForPrefetch();
+
+  olp::client::HRN hrn(catalog);
+  VolatileLayerClient client(hrn, layer, settings_);
+  {
+    SCOPED_TRACE("Prefetch tiles online and store them in memory cache");
+    std::vector<olp::geo::TileKey> tile_keys = {
+        olp::geo::TileKey::FromHereTile(kTileId)};
+
+    {
+      SCOPED_TRACE("min/max levels are 0");
+      auto request = olp::dataservice::read::PrefetchTilesRequest()
+                         .WithTileKeys(tile_keys)
+                         .WithMinLevel(0)
+                         .WithMaxLevel(0);
+
+      std::promise<PrefetchTilesResponse> promise;
+      std::future<PrefetchTilesResponse> future = promise.get_future();
+      auto token = client.PrefetchTiles(
+          request, [&promise](PrefetchTilesResponse response) {
+            promise.set_value(std::move(response));
+          });
+
+      ASSERT_NE(future.wait_for(kTimeout), std::future_status::timeout);
+      PrefetchTilesResponse response = future.get();
+      EXPECT_TRUE(response.IsSuccessful());
+      const auto& result = response.GetResult();
+
+      for (auto tile_result : result) {
+        EXPECT_TRUE(tile_result->IsSuccessful());
+        ASSERT_TRUE(tile_result->tile_key_.IsValid());
+      }
+
+      ASSERT_EQ(tile_keys.size(), result.size());
+    }
+
+    {
+      SCOPED_TRACE(" min level greater than max level");
+      auto request = olp::dataservice::read::PrefetchTilesRequest()
+                         .WithTileKeys(tile_keys)
+                         .WithMinLevel(-1)
+                         .WithMaxLevel(0);
+
+      std::promise<PrefetchTilesResponse> promise;
+      std::future<PrefetchTilesResponse> future = promise.get_future();
+      auto token = client.PrefetchTiles(
+          request, [&promise](PrefetchTilesResponse response) {
+            promise.set_value(std::move(response));
+          });
+
+      ASSERT_NE(future.wait_for(kTimeout), std::future_status::timeout);
+      PrefetchTilesResponse response = future.get();
+      EXPECT_TRUE(response.IsSuccessful());
+      const auto& result = response.GetResult();
+
+      for (auto tile_result : result) {
+        EXPECT_TRUE(tile_result->IsSuccessful());
+        ASSERT_TRUE(tile_result->tile_key_.IsValid());
+      }
+
+      ASSERT_EQ(tile_keys.size(), result.size());
```

Comment:
Due to min level is greater than max this must fail but there are checks for successful fetch. Is this correct?

Reply:
min/max range is invalid if `min>max`, for such case we load only tiles from the request.
Actually, 0 and values > TileKey::MaxLevel(31) are invalid, so this test more like checking invalid levels behavior.
I will add test with valid levels but min>max :-)

Reply:
done.